### PR TITLE
fix(datastore): optional hasOne #10685

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Issue #, if available

#10685 

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by iartemiev — CHANGES_REQUESTED** [review]
Looks good, but lets add a test for this please

**Comment by svidgen** [comment]
> Looks good, but lets add a test for this please

If possible, add to `commonAdapterTests`.

File: packages/datastore/__tests__/helpers.ts [hunk 1]
```diff
@@ -1294,10 +1300,6 @@ export declare class DefaultPKParent {
 }
 
 export declare class DefaultPKChild {
-	readonly [__modelMeta__]: {
-		identifier: OptionallyManagedIdentifier<DefaultPKChild, 'id'>;
-		readOnlyFields: 'createdAt' | 'updatedAt';
-	};
```
Comment (svidgen) [inline packages/datastore/__tests__/helpers.ts:1300]: ?

File: packages/datastore/__tests__/commonAdapterTests.ts [hunk 2]
```diff
@@ -548,6 +561,41 @@ export function addCommonQueryTests({
 				).toBeUndefined();
 			}
 		);
+
+		(isSQLiteAdapter() ? test.skip : test)(
```
Comment (svidgen) [inline packages/datastore/__tests__/commonAdapterTests.ts:565]: Confused. If this is for default PK hasone/belongsto, shouldn't that work on SQLite?

File: packages/datastore/__tests__/helpers.ts (hunk 1 above)
Comment (dpilch) [reply packages/datastore/__tests__/helpers.ts:1300]: Mistake, will fix

File: packages/datastore/__tests__/commonAdapterTests.ts (hunk 2 above)
Comment (dpilch) [reply packages/datastore/__tests__/commonAdapterTests.ts]: Hasty copy and paste job, updated.

Comment (svidgen) [reply packages/datastore/__tests__/commonAdapterTests.ts:565]: As discussed offline, SQLiteAdapter *does* have feature gaps around cascading delete. The first of these two tests should actually pass. But, makes enough sense to just re-enable both when we shore up the adapter.

**Review by svidgen — APPROVED** [review]

**Review by iartemiev — APPROVED** [review]

**Pull request merged**